### PR TITLE
SQL: Handle aggregation for null group

### LINKED ISSUE #34896: SQL: MIN/MAX/AVG/SUM/PERCENTILE/PERCENTILE RANK handling on NULL values
```
sql> select max(languages) max, min(languages) min, sum(languages) sum, percentile(languages, 80) percent, percentile_rank(languages, 3) percent_rank, avg(languages) avg from test_emp group by languages;
      max      |      min      |      sum      |    percent    | percent_rank  |      avg
---------------+---------------+---------------+---------------+---------------+---------------
-Infinity      |Infinity       |0.0            |NaN            |NaN            |NaN
1.0            |1.0            |15.0           |1.0            |100.0          |1.0
2.0            |2.0            |38.0           |2.0            |100.0          |2.0
3.0            |3.0            |51.0           |3.0            |100.0          |3.0
4.0            |4.0            |72.0           |4.0            |0.0            |4.0
5.0            |5.0            |105.0          |5.0            |0.0            |5.0
```

The first line is for `null` values in the `languages` field. Shouldn't MIN and MAX of null values return a `null`? Same question for SUM. Not so sure about PERCENTILE and PERCENTILE_RANK though... ES returns `null` for these two. If MIN and MAX would return `null` probably the same should be done for AVG.
Opening this one up for discussion.

### PR DESCRIPTION
**Pull request by costin**
When dealing with a null group, make sure the associated aggs return
null

Fix #34896

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java [hunk 1]
```diff
@@ -79,6 +89,33 @@ public Object extract(Bucket bucket) {
         return innerKey != null && v instanceof Map ? ((Map<?, ?>) v).get(innerKey) : v;
     }
 
+    /**
+     * Check if the given aggregate has been executed and has computed values
+     * or not (the bucket is null).
+     * 
+     * Hopefully solved by
```
Comment (astefan) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java]: Incomplete statement?

**Review by astefan — APPROVED** [review]
LGTM

**Review by matriv — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java]: Please complete the sentence ;-)

**Pull request merged**